src/Provider/Pool.php

```diff
@@ -157,31 +159,19 @@ public function getContexts(): array
     }
 
     /**
-     * @return string[]|null
+     * @return string[]
      */
-    public function getProviderNamesByContext(string $name): ?array
+    public function getProviderNamesByContext(string $name): array
```

Comment:
Here I assume to the user calls the hasContext previously (same for other methods). wdyt @VincentLanglet ?

Reply:
good idea